posydon/utils/common_functions.py

```diff
@@ -195,52 +195,54 @@ def rzams(m, z=0.02, Zsun=0.02):
     return r
 
 
-'''
-
-
-Receives:
-q ->
-a_orb ->
-
-Returns:
-RL -> Roche lobe radius in similar units as a_orb
-'''
-
-
 def roche_lobe_radius(m1, m2, a_orb=1):
     """Approximate the Roche lobe radius from [1]_.
 
     Parameters
     ----------
-    m1 : float
+    m1 : float, array of floats
         the mass of the star for which we calculate the Roche lobe
-    m2: float
+    m2 : float, array of floats
         the mass of the companion star
-    a_orb : float
+    a_orb : float, array of floats
         Orbital separation. The return value will have the same unit.
 
     Returns
     -------
-    float
+    float, array of floats
         Roche lobe radius in similar units as a_orb
     References
     ----------
     .. [1] Eggleton, P. P. 1983, ApJ, 268, 368
 
     """
-
-    if isinstance(a_orb, np.ndarray):
+    
+    ## catching if a_orb is an empty array or is an array with invalid separation values
+    if isinstance(a_orb, np.ndarray) and (not np.any(a_orb) or np.any(a_orb <=0)):
+        Pwarn("Trying to compute RL radius for binary with no separation", "EvolutionWarning")
+        a_orb = np.full_like(a_orb, np.nan)
```

Comment:
If you like to only set input values to nan. I'd suggest to only set the inappropriate ones to nan, hence if there is only one bad value in an array, you can get all other values correctly calculated.